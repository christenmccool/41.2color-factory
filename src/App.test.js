import { render, fireEvent } from '@testing-library/react';
import App from './App';
import {MemoryRouter} from 'react-router-dom';


test('renders without crashing', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

test("if matches snapshot", () => {
  const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);

  expect(asFragment()).toMatchSnapshot();
})

test("if renders home", () => {
  const {queryByText, debug} = render(<MemoryRouter><App /></MemoryRouter>);

  expect(queryByText("Welcome to the Color Factory.")).toBeInTheDocument();
  expect(queryByText("Please select a color.")).toBeInTheDocument();
})

test("if add a color link works", () => {
  const {queryByText, debug} = render(<MemoryRouter><App /></MemoryRouter>);
  const link = queryByText("Add a color");

  fireEvent.click(link);

  expect(queryByText("Welcome to the Color Factory.")).not.toBeInTheDocument();
  expect(queryByText("Please select a color.")).not.toBeInTheDocument();

  expect(queryByText("Color Name")).toBeInTheDocument();
  expect(queryByText("Color Value")).toBeInTheDocument();

})

test("if blue link works", () => {
  const {queryByText, debug} = render(<MemoryRouter><App /></MemoryRouter>);
  const link = queryByText("blue");

  fireEvent.click(link);

  expect(queryByText("Welcome to the Color Factory.")).not.toBeInTheDocument();
  expect(queryByText("Please select a color.")).not.toBeInTheDocument();

  expect(queryByText("This is blue")).toBeInTheDocument();

})

test("if add a color works", () => {
  const {queryByLabelText,queryByText, debug} = render(<MemoryRouter initialEntries={['/colors/new']}><App /></MemoryRouter>);

  const nameInput = queryByLabelText("Color Name");
  const hexInput = queryByLabelText("Color Value");
  const link = queryByText("Add this color");

    expect(queryByText("yellow")).not.toBeInTheDocument();

  fireEvent.change(nameInput, {target: {value: "yellow"}});
  fireEvent.change(hexInput, {target: {value: "#FFFF00"}});

  fireEvent.click(link);
  expect(queryByText("yellow")).toBeInTheDocument();

})


