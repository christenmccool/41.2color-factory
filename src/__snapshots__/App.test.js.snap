// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ColorList"
    >
      <div
        class="ColorList-top"
      >
        <h3>
          Welcome to the Color Factory.
        </h3>
        <a
          class="ColorList-addBtn"
          href="/colors/new"
        >
          Add a color
        </a>
      </div>
      <div
        class="ColorList-main"
      >
        <h3>
          Please select a color.
        </h3>
        <a
          href="/colors/blue"
        >
          blue
        </a>
        <a
          href="/colors/green"
        >
          green
        </a>
        <a
          href="/colors/orange"
        >
          orange
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
